packages/ckeditor5-engine/tests/model/utils/insertcontent.js

```diff
@@ -1299,128 +1302,1629 @@ describe( 'DataController utils', () => {
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0, 2 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 0, 5 ] );
+				} );
+
+				it( 'should calculate affected range correctly after inserting paragraph before another one', () => {
+					setData( model, '[]<paragraph>foo</paragraph>' );
+
+					const affectedRange = insertHelper( '<paragraph>Bar</paragraph>', {
+						'marker-a': { start: [ 0, 0 ], end: [ 0, 3 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0, 0 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 0, 3 ] );
+				} );
+
+				it( 'should return affected range correctly after inserting content with multiple markers', () => {
+					setData( model, '<paragraph>foo[]</paragraph>' );
+
+					const affectedRange = insertHelper( '<paragraph>Bar</paragraph>', {
+						'marker-a': { start: [ 0, 0 ], end: [ 0, 1 ] },
+						'marker-b': { start: [ 0, 1 ], end: [ 0, 2 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0, 3 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 0, 6 ] );
+				} );
+
+				it( 'should calculate affected range correctly after inserting content with nested markers', () => {
+					setData( model, '<paragraph>foo[]</paragraph>' );
+
+					const affectedRange = insertHelper( '<paragraph>Bar</paragraph>', {
+						'marker-a': { start: [ 0, 0 ], end: [ 0, 3 ] },
+						'marker-b': { start: [ 0, 1 ], end: [ 0, 2 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0, 3 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 0, 6 ] );
+				} );
+
+				it( 'should calculate affected range correctly after inserting imageBlock', () => {
+					setData( model, '<paragraph>foo[]</paragraph>' );
+
+					const affectedRange = insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 1 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 1 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 2 ] );
+				} );
+
+				it( 'should calculate affected range correctly after inserting paragraph in empty content', () => {
+					setData( model, '' );
+
+					const affectedRange = insertHelper( '<paragraph>Bar</paragraph>', {
+						'marker-a': { start: [ 0, 0 ], end: [ 0, 3 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 1 ] );
+				} );
+
+				it( 'should calculate affected range correctly after inserting imageBlock in empty content', () => {
+					setData( model, '' );
+
+					const affectedRange = insertHelper( '<imageBlock></imageBlock>', {
+						'marker-a': { start: [ 0 ], end: [ 1 ] }
+					} );
+
+					expect( affectedRange.start.path ).to.deep.equal( [ 0 ] );
+					expect( affectedRange.end.path ).to.deep.equal( [ 1 ] );
+				} );
+			} );
+		} );
+	} );
+
+	describe( 'integration with limit elements', () => {
```

Comment:
Just to confirm: you didn't change anything here but the GH diff got it incorrectly?

Reply:
That's true. I didn't change those tests. Screenshot:

![](https://user-images.githubusercontent.com/20561161/192485510-6e344f07-b4cd-4061-9a4d-04c7951f1013.png)